### PR TITLE
MM-35191 - show banner in user timezone

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
Show cloud trial banner last day notification date in user local date timezone.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35191

https://mattermost.atlassian.net/browse/MM-36328

#### Screenshots
Captura de pantalla 2021-06-09 a las 20 49 32
Captura de pantalla 2021-06-09 a las 20 50 22

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -268,7 +268,7 @@
   "admin.billing.subscription.cancelSubscriptionSection.description": "At this time, deleting a workspace can only be done with the help of a customer support representative.",
   "admin.billing.subscription.cancelSubscriptionSection.title": "Cancel your subscription",
   "admin.billing.subscription.cloudTrial.daysLeftOnTrial": "There are {daysLeftOnTrial} days left on your free trial",
-  "admin.billing.subscription.cloudTrial.lastDay": "This is the last day of your free trial. Your access will expire today at 11:59pm PT.",
+  "admin.billing.subscription.cloudTrial.lastDay": "This is the last day of your free trial. Your access will expire at {userEndTrialHour} PT.",
```
Comment (marianunez) [inline i18n/en.json:271]: We should add the users timezone in the time/date formatting and remove this hardcoded `Pacific Time` reference
```suggestion
  "admin.billing.subscription.cloudTrial.lastDay": "This is the last day of your free trial. Your access will expire at {userEndTrialHour}.",
```

Comment (pablovelezvidal) [reply i18n/en.json:271]: ok!

**Comment by pablovelezvidal** [comment]
Captura de pantalla 2021-06-10 a las 12 47 47

**Review by marianunez — APPROVED** [review]
Great work @pablovelezvidal!

**Comment by stevemudie** [comment]
@pablovelezvidal Nice work! This is really shaping up.

I do have a rewording suggestion and a question though.

Right now the banner reads (for example) "Your access will expire at June 11th 2021, 18:00:00 EDT."
Typically though we refer to a date with "on" and a time with "at". With that in mind it would read better if it was formatted as "Your access will expire **on** June 11th 2021 **at** 18:00:00 EDT."

Or, since it's the last day of the trial would it be possible to just eliminate the date? If so then "Your access will expire at 18:00:00 EDT." works as is since the full banner line (with the date eliminated) reads: "This is the last day of your free trial. Your access will expire at 18:00:00 EDT."

As for my question, I noticed that when I adjust my system time and the timezone settings in Mattermost it drops the timezone abbreviation completely. If I leave it at EDT (New York) then it reads EDT no matter where in the world I set my system timezone to. Any ideas?

**Comment by pablovelezvidal** [comment]
> @pablovelezvidal Nice work! This is really shaping up.
> 
> I do have a rewording suggestion and a question though.
> 
> Right now the banner reads (for example) "Your access will expire at June 11th 2021, 18:00:00 EDT."
> Typically though we refer to a date with "on" and a time with "at". With that in mind it would read better if it was formatted as "Your access will expire **on** June 11th 2021 **at** 18:00:00 EDT."
> 
> Or, since it's the last day of the trial would it be possible to just eliminate the date? If so then "Your access will expire at 18:00:00 EDT." works as is since the full banner line (with the date eliminated) reads: "This is the last day of your free trial. Your access will expire at 18:00:00 EDT."
> 
> As for my question, I noticed that when I adjust my system time and the timezone settings in Mattermost it drops the timezone abbreviation completely. If I leave it at EDT (New York) then it reads EDT no matter where in the world I set  system timezone to. Any ideas?

I think we should keep the date, because showing just the time could lead to confusions, since the trial ending could happen at 14:00 of tomorrow, and user can be in the 15:00 of Today and yet still see the banner. Obviously is clear that it means the 14:00 of tomorrow, but sometimes is not that obvious and let's avoid the user having to have to figure it out. Other option, and I think it is possible is to show "Today" or "Tomorrow" but also potentially error prone, date/time management UX is always complex. So, my conclusion is, I agree to keep the date with the wording suggestion from Steve using on for the date and at for the time (4/5). @marianunez thoughts?

File: components/admin_console/billing/billing_summary/billing_summary.tsx [hunk 2]
```diff
@@ -94,13 +94,13 @@ export const freeTrial = (onUpgradeMattermostCloud: () => void, daysLeftOnTrial:
             <img src={upgradeMattermostCloudImage}/>
         </div>
         <div className='UpgradeMattermostCloud__title'>
-            {daysLeftOnTrial !== TrialPeriodDays.TRIAL_1_DAY &&
+            {(daysLeftOnTrial !== TrialPeriodDays.TRIAL_1_DAY && daysLeftOnTrial !== TrialPeriodDays.TRIAL_0_DAYS) &&
```
Comment (nickmisasi) [inline components/admin_console/billing/billing_summary/billing_summary.tsx:97]: Could you simplify this to `daysLeftOnTrial > TrialPeriodDays.TRIAL_1_DAY`?

**Review by nickmisasi — APPROVED** [review]
One suggestion, otherwise, LGTM!

**Review by stevemudie — APPROVED** [review]
Looks good and worked as expected.

**Comment by marianunez** [comment]
> I think we should keep the date, because showing just the time could lead to confusions, since the trial ending could happen at 14:00 of tomorrow, and user can be in the 15:00 of Today and yet still see the banner. Obviously is clear that it means the 14:00 of tomorrow, but sometimes is not that obvious and let's avoid the user having to have to figure it out. Other option, and I think it is possible is to show "Today" or "Tomorrow" but also potentially error prone, date/time management UX is always complex. So, my conclusion is, I agree to keep the date with the wording suggestion from Steve using on for the date and at for the time (4/5). @marianunez thoughts?

+1 on keeping the date but let's adjust the wording so that is properly using `at` vs `on` as suggested.

**Review by annelieseklein — APPROVED** [review]

**Comment by pablovelezvidal** [comment]
Captura de pantalla 2021-06-24 a las 14 42 13

**Comment by stevemudie** [comment]
Tested this out on https://github.com/mattermost/mattermost-webapp/pull/8308 and it worked as expected.

**Pull request merged**